pybandits/utils.py

```diff
@@ -42,3 +68,62 @@ def extract_argument_names_from_function(function_handle: Callable, is_class_met
     start_index = int(is_class_method)
     argument_names = function_handle.__code__.co_varnames[start_index : function_handle.__code__.co_argcount]
     return argument_names
+
+
+def in_jupyter_notebook() -> bool:
+    """
+    Check if the code is running in a Jupyter notebook.
+
+    Reference: https://stackoverflow.com/a/39662359
+
+    Returns
+    -------
+    bool
+        True if the code is running in a Jupyter notebook, False otherwise.
+    """
+
+    try:
+        shell = get_ipython().__class__.__name__
+
+        if shell == "ZMQInteractiveShell":
+            return True  # Jupyter notebook or qtconsole
+
+        elif shell == "TerminalInteractiveShell":
+            return False  # Terminal running IPython
+
+        else:
+            return False  # Other type (likely shouldn't happen)
```

Comment:
instead of returning False, Shall we raise an exception? @adarmiento wdyt?

Reply:
I Agree

Reply:
On second thought, I did another change, but now this utility function simply matches its name -> True for Jupyter, False otherwise. I don't see reason to raise exception on the else, as the else can happen for many scenarios (each associated with different shell value): console, UTs, debug mode, and more...